src/targets/crates.ts

```diff
@@ -227,9 +255,26 @@ export class CratesTarget extends BaseTarget {
       crate.manifest_path
     );
 
-    return spawnProcess(CARGO_BIN, args, {
-      env: { ...process.env, CARGO_REGISTRY_TOKEN: this.cratesConfig.apiToken },
-    });
+    const env = {
+      ...process.env,
+      CARGO_REGISTRY_TOKEN: this.cratesConfig.apiToken,
+    };
+
+    let delay = RETRY_DELAY_SECS;
+    logger.info(`Publishing ${crate.name}`);
+    for (let i = 0; i <= MAX_ATTEMPTS; i++) {
+      try {
+        return await spawnProcess(CARGO_BIN, args, { env });
+      } catch (e) {
+        if (i < MAX_ATTEMPTS && e.message.includes(VERSION_ERROR)) {
```

Comment:
This differs from the for loop's condition, intentional? I find having two checks a bit confusing btw. Feels like this should be a while loop.

```js
const MAX_WAIT_SECS = 60;
let totalWait = 0;
let delay = RETRY_DELAY_SECS;

while (totalWait <= MAX_WAIT_SECS) {
  try {
    return await spawnProcess(CARGO_BIN, args, { env });
  catch (err) {
    if (!e.message.includes(VERSION_ERROR)) {
      throw err;
    }

    logger.warn(`Publish failed due to potentially stale cache. Trying again in ${delay}s...`);
    await sleepAsync(delay * 1000);
    totalWait += delay;
    delay *= RETRY_EXP_FACTOR;
  }
}
```

Reply:
Yes this is intentional. The loop needs to to take another turn so that we get into the `else` branch. In your implementation, you do not throw the error in the last iteration.

I thought about a total wait time (which could be written as a `for` loop, too), but rather wanted a fixed number of retries. And then doing `RETRY_DELAY_SECS * Math.pow(RETRY_EXP_FACTOR, MAX_ATTEMPTS)` seemed to cumbersome.

Reply:
>In your implementation, you do not throw the error in the last iteration.

Yes I do as I do not check the count or total wait in the if separately.